phpdotnet/phd/Highlighter.php

```diff
@@ -63,7 +63,10 @@ public function highlight($text, $role, $format)
 
         if ($role == 'php') {
             try {
-                return str_replace('&nbsp;', ' ', highlight_string($text, 1));
```

Comment:
``str_replace()`` can also take an array of values to replace and an array of what those values should be, why not just use that?

But the issue seems to be more that ``\n`` tags before and after are being converted to ``<br>`` so wouldn't it make more sense to just ``ltrim()`` new lines of the ``$text`` variable?

Reply:
> `str_replace()` can also take an array of values to replace and an array of what those values should be, why not just use that?

see https://github.com/php/phd/pull/79#discussion_r1287289455

> But the issue seems to be more that \n tags before and after are being converted to <br> so wouldn't it make more sense to just ltrim() new lines of the $text variable?

This wouldn't work, because the extra newlines are inserted by `highlight_string()`, namely here:

https://github.com/php/php-src/blob/f8b42da3a4635aa2ace3d4dbfdbae6835fb43d7f/Zend/zend_highlight.c#L92

and here:

https://github.com/php/php-src/blob/f8b42da3a4635aa2ace3d4dbfdbae6835fb43d7f/Zend/zend_highlight.c#L165-L167